tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+    nexus_param = {
+            'type': api_name + 'nexus', 'hostname': 'nexus-01',
+            'username': 'root', 'password': 'root1234', 'dummy_vlan': '333'
+            }
+    mock_param = {
+            'type': api_name + 'mock', 'hostname': 'mockSwitch-01',
+            'username': 'root', 'password': 'root1234'
+            }
+    brocade_param = {
+            'type': api_name + 'brocade', 'hostname': 'brocade-01',
+            'username': 'root', 'password': 'root1234',
+            'interface_type': 'TenGigabitEthernet'
+            }
+
+    sess.put(url_switch + 'dell-01', data=json.dumps(dell_param))
+    sess.put(url_switch + 'nexus-01', data=json.dumps(nexus_param))
+    sess.put(url_switch + 'mock-01', data=json.dumps(mock_param))
+    sess.put(url_switch + 'brocade-01', data=json.dumps(brocade_param))
+
+    # Adding ports to the mock switch, Connect nics to ports:
+    for i in range(1, 8):
+        sess.put(url_switch + 'mock-01/port/gi1/0/' + repr(i))
+        sess.post(
+                url_switch + 'mock-01/port/gi1/0/' + repr(i) + '/connect_nic',
+                data=json.dumps(
+                    {'node': 'node-0' + repr(i), 'nic': 'eth0'}
+                    )
+                )
+
+# Adding port gi1/0/8 to switch mock-01 without connecting it to any node.
+    sess.put(url_switch + 'mock-01/port/gi1/0/8')
+
+    # Adding Projects proj-01 - proj-03
+    for i in ["proj-01", "proj-02", "proj-03"]:
+        sess.put('http://127.0.0.1:8000/project/' + i)
+
+    # Allocating nodes to projects
+    url_project = 'http://127.0.0.1:8000/project/'
+    # Adding nodes 1 to proj-01
+    sess.post(
+            url_project + 'proj-01' + '/connect_node',
+            data=json.dumps({'node': 'node-01'})
+            )
+    # Adding nodes 2, 4 to proj-02
+    sess.post(
+            url_project + 'proj-02' + '/connect_node',
+            data=json.dumps({'node': 'node-02'})
+            )
+    sess.post(
+            url_project + 'proj-02' + '/connect_node',
+            data=json.dumps({'node': 'node-04'})
+            )
+    # Adding node  3, 5 to proj-03
+    sess.post(
+            url_project + 'proj-03' + '/connect_node',
+            data=json.dumps({'node': 'node-03'})
+            )
+    sess.post(
+            url_project + 'proj-03' + '/connect_node',
+            data=json.dumps({'node': 'node-05'})
+            )
+
+    # Assigning networks to projects
+    url_network = 'http://127.0.0.1:8000/network/'
+    for i in ['net-01', 'net-02', 'net-03']:
+        sess.put(
+                url_network + i,
+                data=json.dumps(
+                    {"owner": "proj-01", "access": "proj-01", "net_id": ""}
+                    )
+                )
+
+    for i in ['net-04', 'net-05']:
+        sess.put(
+                url_network + i,
+                data=json.dumps(
+                    {"owner": "proj-02", "access": "proj-02", "net_id": ""}
+                    )
+                )
+
+
+# -- SETUP --
```

Comment:
While developing tests It helped to jump to different sections of file between, config, populate and setup. Removed it.